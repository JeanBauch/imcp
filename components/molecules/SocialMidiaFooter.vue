<template>
	<div class="flex justify-around my-10">
		<div v-for="(social, index) in content" :key="index">
			<ButtonSocialMidia :social-midia="social.socialMidia" :link="''" />
		</div>
	</div>
</template>

<script lang="ts">
import Vue from 'vue';
import ButtonSocialMidia from '../atoms/ButtonSocialMidia.vue';

declare interface SocialMida {
	socialMidia: string,
	link: string
}

declare interface contentType {
	content: SocialMida[]
}

export default Vue.extend({
	data ():contentType {
		return {
			content: [
				{
					socialMidia: 'facebook',
					link: '',
				},
				{
					socialMidia: 'instagram',
					link: '',
				},
				{
					socialMidia: 'spotify',
					link: '',
				},
				{
					socialMidia: 'youtube',
					link: '',
				},
			],
		};
	},
	components: { ButtonSocialMidia },
});
</script>
