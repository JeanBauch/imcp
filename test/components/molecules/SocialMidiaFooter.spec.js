import { shallowMount } from '@vue/test-utils';

import SocialMidiaFooter from '~/components/molecules/SocialMidiaFooter.vue';

describe('Components:organisms:Footer', () => {
	const buildWrapper = () => {
		return {
		};
	};
	describe('Snapshots', () => {
		it('should render component', () => {
			const config = buildWrapper();
			const wrapper = shallowMount(SocialMidiaFooter, config);

			expect(wrapper.html()).toMatchSnapshot();
		});
	});
});
