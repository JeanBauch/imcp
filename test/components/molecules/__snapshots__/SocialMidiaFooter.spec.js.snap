// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components:organisms:Footer Snapshots should render component 1`] = `
"<div class=\\"flex justify-around my-10\\">
  <div>
    <buttonsocialmidia-stub link=\\"\\" socialmidia=\\"facebook\\"></buttonsocialmidia-stub>
  </div>
  <div>
    <buttonsocialmidia-stub link=\\"\\" socialmidia=\\"instagram\\"></buttonsocialmidia-stub>
  </div>
  <div>
    <buttonsocialmidia-stub link=\\"\\" socialmidia=\\"spotify\\"></buttonsocialmidia-stub>
  </div>
  <div>
    <buttonsocialmidia-stub link=\\"\\" socialmidia=\\"youtube\\"></buttonsocialmidia-stub>
  </div>
</div>"
`;
